pkg/controller/remotemachineset/remotemachineset_controller_test.go

```diff
@@ -311,6 +313,37 @@ func TestRemoteMachineSetReconcile(t *testing.T) {
 				}
 			}(),
 		},
+		{
+			// Can't control what a user might do in their cluster.
+			name: "Nil ProviderSpec",
+			localExisting: []runtime.Object{
+				testClusterDeployment([]hivev1.MachinePool{
+					testMachinePool("worker", 3, []string{}),
+				}),
+				testSecret(adminKubeconfigSecret, adminKubeconfigSecretKey, testName),
+				testSecret(adminPasswordSecret, adminPasswordSecretKey, testName),
+				testSecret(sshKeySecret, sshKeySecretKey, testName),
+			},
+			remoteExisting: []runtime.Object{
+				func() *machineapi.MachineSet {
+					ms := testMachineSet("foo-12345-worker-us-east-1c", "worker", true, 1, 0)
+					ms.Spec.Template.Spec.ProviderSpec.Value = nil
+					return ms
+				}(),
+			},
+			expectedRemoteMachineSets: func() *machineapi.MachineSetList {
```

Comment:
We don't need this. It will never be used since the test will short-circuit after checking the returned error.

Reply:
removed